packages/ove-service-auth/README.md

```diff
@@ -0,0 +1,191 @@
+
+#### Important note on proxies
+
+The proxy mapping rules are overridden by the rest service mappings. In case authentication and locking are enabled
+the proxy will serve these routes instead of proxying the requests. Currently all the auth operations are performed 
+under **auth/login** and **auth/lock** routes.
+
+### Authentication
+
+This feature allows users to authenticate with standard username and password. The authentication is performed using 
+a standard jwt protocol. The user config, with username and passwords are described in the **config/users.json** file:
+
+```json
+[
+  {
+    "id": "admin",
+    "password": "secure-password-123"
+  }
+]
+``` 
+
+**Note:** The user id has to be unique.
+
+All the other authentication parameters can be set using environment variables:
+
+- **SERVICE_LOGIN_ENABLED**: enable the login feature, default is True.
+- **SERVICE_LOGIN_CONFIG**: location of the login configuration file, containing the usernames and passwords. The default 
+location is  **config/users.json**.
+- **SERVICE_LOGIN_HASH_PASSWORDS**: enable the password hashing feature after loading from file, default is True. 
+Must be enabled if the passwords are stored in plain text.
+- **JWT_TOKEN**: a secure token to be used by the JWT algorithm. By default a secure random password is generated.
+- **JWT_ISSUER**: the name of the issuer to be used by the JWT algorithm, default is heimdall. This can be personalized 
+to something more secure.
+- **JWT_TOKEN_EXPIRATION**: JWT token expiration parameter (in seconds), default is 3600. 
+
+### Authorization
+
+Authorization can be configured for every route by using a read or write token. These can be configured in 
+**config/access.json**: 
+
+```json
+{
+  "read": [
+    "very-secure-token"
+  ],
+  "write": [
+    "slightly-more-secure-token-because-it's-longer"
+  ]
+}
+```
+
+When authorization is enabled, the user can be given access via a token by adding this information
+to the user profile in the **config/users.json** file:
+
+```json
+[
+  {
+    "id": "admin",
+    "password": "secure-password-123",
+    "access_tokens": [
+      "slightly-more-secure-token-because-it's-longer"
+    ]
+  }
+]
+``` 
+
+All the other authorization parameters can be set using environment variables:
+
+- **SERVICE_ACCESS_ENABLED**: enable the authorization feature, default is True.
+- **SERVICE_ACCESS_CONFIG**: location of the access configuration file, containing the authorization tokens. The default 
+location is  **config/users.json**.
+
+### Locking
+
+API Locking can be performed for the whole API to restrict temporarily the access. All the lock settings can be configured
+using environment variables:
+
+- **SERVICE_LOCK_ENABLED**: enable the lock feature, default is True.
+- **SERVICE_LOCK_KEY_LENGTH**: length of the lock key, default is 8. The size should be kept to a reasonable low value
+to make the feature more user friendly.
```

Comment:
```suggestion
to make the feature more user friendly. The purpose of this feature is to prevent accidental API calls to a space that someone else is using, rather than to provide security.
```